### PR TITLE
PGIF Logging

### PR DESCRIPTION
**Pull request by kenshen112**
### Description of Changes

Adds the option to turn pgif logs on or off and adds extra missing error reports

### Rationale behind Changes

pgif unrestrained tends to eat the console hiding any other useful output

### Suggested Testing Steps

play a game you know eats ass on pgif errors and try with it off

### CODE REVIEW

File: pcsx2/HwWrite.cpp [hunk 1]
```diff
@@ -26,6 +26,11 @@
 #include "SPU2/spu2.h"
 #include "R3000A.h"
 
+#include "App.h"
+#include "ConsoleLogger.h"
+#include "DebugTools/Debug.h"
+#include "Utilities/Console.h"
+
```
Comment (refractionpcsx2) [inline pcsx2/HwWrite.cpp:33]: does it really need all of those for 1 log call?

File: pcsx2/gui/ConsoleLogger.cpp [hunk 2]
```diff
@@ -324,17 +325,16 @@ static const bool ConLogDefaults[] =
...
 	false,
 	false,
 	false,
 	false,
+	true,
 #ifndef DISABLE_RECORDING
 	false,
-	false,
-#endif
 	false
```
Comment (refractionpcsx2) [inline pcsx2/gui/ConsoleLogger.cpp:336]: why were these removed?

File: pcsx2/ps2/pgif.cpp [hunk 3]
```diff
@@ -1063,7 +1052,7 @@ void dma6_OTClear()
     //u32 blkCnt = (psxHu(0x1f8010e4) >> 16) & 0xFFFF;
     u32 addrPnt = madr;
     u32 pntCnt = 0;
-    Console.WriteLn("DMA6 OT clr madr %08X  bcr %08X  chcr %08X", psxHu(0x1f8010e0), psxHu(0x1f8010e4), psxHu(0x1f8010e8));
+    pgifConLog("DMA6 OT clr madr %08X  bcr %08X  chcr %08X", psxHu(0x1f8010e0), psxHu(0x1f8010e4), psxHu(0x1f8010e8));
```
Comment (refractionpcsx2) [inline pcsx2/ps2/pgif.cpp:1056]: Is this even related to pgif?

File: pcsx2/gui/ConsoleLogger.cpp (hunk 2 above)
Comment (kenshen112) [reply pcsx2/gui/ConsoleLogger.cpp:336]: Sorry was experimenting and forgot about that. I'll fix that in a bit

File: pcsx2/ps2/pgif.cpp (hunk 3 above)
Comment (kenshen112) [reply pcsx2/ps2/pgif.cpp:1056]: idk I can clear it if you want

Comment (refractionpcsx2) [reply pcsx2/ps2/pgif.cpp:1056]: Well, if it's not pgif related, there's no need to mark it as pgif, that's just confusing

**Comment by Mrlinkwii** [comment]
From my brief testing the  PR works good :+1:

**Comment by refractionpcsx2** [comment]
Can you squash, and also can you put // before your comment and not rely on the commented out line.

**Comment by kenshen112** [comment]
> 
> 
> Can you squash, and also can you put // before your comment and not rely on the commented out line.

Done.
